tests/components/homekit/test_get_accessories.py

```diff
@@ -38,16 +39,21 @@ def test_customize_options(config, name):
 
 @pytest.mark.parametrize('type_name, entity_id, state, attrs, config', [
     ('Fan', 'fan.test', 'on', {}, {}),
+
     ('Light', 'light.test', 'on', {}, {}),
+
     ('Lock', 'lock.test', 'locked', {}, {ATTR_CODE: '1234'}),
 
+    ('MediaPlayer', 'media_player.test', 'on', {}, {CONF_MODE: 'on_off'}),
```

Comment:
We need an extra test case for when the `media_player` should not be added. While thinking about it: What are the use cases for this / the user input (for `entity_config/mode`) to reach it.

Reply:
You're right, now that we added this to `validate_media_player_modes`
```python
    if not config[CONF_MODE]:
        config[CONF_MODE] = supported_modes
        return

    for mode in config[CONF_MODE]:
        if mode not in supported_modes:
            raise vol.Invalid('"{}" does not support mode: "{}".'
                              .format(state.entity_id, mode))
```
Then this in `get_accessory`
```python
    elif state.domain == 'media_player':
        validate_media_player_modes(state, config)
        if config[CONF_MODE]:
            a_type = 'MediaPlayer'
```
Can be re-written as 
```python
    elif state.domain == 'media_player':
        validate_media_player_modes(state, config)
        a_type = 'MediaPlayer'
```
I also added a test in `test_not_supported` in the event that the `media_player` entity has no supported modes (i.e. [Universal](https://www.home-assistant.io/components/media_player.universal/)) .
```python
    with pytest.raises(vol.Invalid):
        attrs = {ATTR_SUPPORTED_FEATURES: SUPPORT_PAUSE | SUPPORT_SEEK}
        entity_state = State('media_player.demo', 'on', attrs)
        get_accessory(None, entity_state, 2, {CONF_MODE: [ON_OFF]})
```

Reply:
I would need to see the final version for this. If there are entities that don't match any supported modes, we might as well need the change `if config[CONF_MODE]: a_type = 'MediaPlayer'`. Could be possible that I'm missing something.

Reply:
If an entity has no supported modes, then `supported_modes` will be empty and a `vol.Invalid` will be raised from `if mode not in supported_modes:`. Anyway, I pushed my latest version so you can review.